docs/versioning.rst

```diff
@@ -0,0 +1,145 @@
+**********
+Versioning
+**********
+
+
+Current Release Stages
+======================
+
+.. list-table:: Releases
+   :widths: 20 30 30 20
+   :header-rows: 1
+
+   * - Release
+     - Runtimes
+     - Minimum Trace Agent
+     - Stage
+   * - 0.x
+     - CPython: 2.7, 3.5–3.10
+     - 
+     - MAINTENANCE [2022-03-01]
+   * - 1.x
+     - CPython: 2.7, 3.5+
+     - 7.28
+     - CURRENT
+
+
+Runtimes
+========
+
+
+.. list-table:: Supported runtime versions
+   :widths: 60 40
+   :header-rows: 1
+
+   * - Runtime
+     - Runtime version
+   * - CPython 
+     - 3.10
+     - 3.9
+     - 3.8
+     - 3.7
+     - 3.6
+     - 2.7
+
+   
+.. list-table:: Supported operating system
+   :widths: 60 40
+   :header-rows: 1
+
+   * - Operating system 
+     - CPU architecture
+     - Tested in CI
+   * - Linux
+     - x86-64
+     - :white_check_mark:
+   * - Linux
+     - i686
+   * - Linux
+     - AArch64
+   * - MacOS
+     - Intel
+   * - MacOS
+     - Apple Silicon
+   * - Windows
+     - 64bit
+   * - Windows
+     - 32bit
+
+
+Interfaces
+==========
+
+The definition of the **internal** interface:
+    The ``ddtrace.internal`` package is internal
+
+    The ``ddtrace.vendor`` package is internal
+
+    Any module, function, class or attribute that is prefixed with a single underscore is internal
+
+    Any package, module, function, class or attribute that is contained within an internal namespace is also internal
+
+
+The definition of the **public** interface:
+    Any package, module, function, class or attribute that is not internal
+
+
+
+Versions
+========
+
+The non-negative integer components of the **version format** (``v<MAJOR>.<MINOR>.<PATCH>``) are incremented by the criteria:
+
+MAJOR
+    Incompatible changes to the public interface
+
+    Dropping support for a Python runtime
```

Comment:
```suggestion
    Dropping support for a Python runtime version
```

? do we need to clarify this? does "Runtime version" mean "CPython" or "CPython 3.7" ?

Reply:
```suggestion
    Removing support for a Python runtime
```

?